crates/iota-light-client/src/checkpoint.rs

```diff
@@ -0,0 +1,672 @@
+        // directory
+        let summary_path = config.checkpoint_summary_file_path(seq);
+
+        // If file exists read the file otherwise download it from the server
+        let summary = if summary_path.exists() {
+            read_checkpoint_summary(config, seq).context("Failed to read checkpoint summary")?
+        } else {
+            panic!("corrupted checkpoint directory");
+        };
+
+        // Verify the checkpoint
+        summary.clone().try_into_verified(&prev_committee)?;
+
+        info!(
+            "Verified epoch: {}, checkpoint: {seq}, checkpoint digest: {}",
+            summary.epoch(),
+            summary.digest()
+        );
+
+        // Extract the next committee information
+        if let Some(EndOfEpochData {
+            next_epoch_committee,
+            ..
+        }) = &summary.end_of_epoch_data
+        {
+            let next_committee = next_epoch_committee.iter().cloned().collect();
+            prev_committee =
+                Committee::new(summary.epoch().checked_add(1).unwrap(), next_committee);
+        } else {
+            bail!("Expected all checkpoints to be end-of-epoch checkpoints");
+        }
+    }
+
+    Ok(())
+}
+
+#[derive(Clone, Debug)]
+struct CheckpointSummaryFileStore<'a> {
+    config: &'a Config,
+}
+
+impl<'a> CheckpointSummaryFileStore<'a> {
+    fn new(config: &'a Config) -> Self {
+        Self { config }
+    }
+}
+
+impl WriteStore for CheckpointSummaryFileStore<'_> {
+    fn insert_checkpoint(
+        &self,
+        checkpoint: &VerifiedCheckpoint,
+    ) -> iota_types::storage::error::Result<()> {
+        let path = self
+            .config
+            .checkpoint_summary_file_path(*checkpoint.sequence_number());
+        info!("Downloading checkpoint summary to '{}'", path.display());
+        bcs::serialize_into(
+            &mut fs::File::create(&path).expect("error writing file"),
+            &checkpoint.clone().into_inner(),
+        )
+        .expect("error serializing summary checkpoint to bcs");
+        Ok(())
+    }
+
+    fn update_highest_synced_checkpoint(
+        &self,
+        _: &iota_types::messages_checkpoint::VerifiedCheckpoint,
+    ) -> iota_types::storage::error::Result<()> {
+        unimplemented!()
+    }
+
+    fn update_highest_verified_checkpoint(
+        &self,
+        _: &iota_types::messages_checkpoint::VerifiedCheckpoint,
+    ) -> iota_types::storage::error::Result<()> {
+        unimplemented!()
+    }
+
+    fn insert_checkpoint_contents(
+        &self,
+        _: &iota_types::messages_checkpoint::VerifiedCheckpoint,
+        _: iota_types::messages_checkpoint::VerifiedCheckpointContents,
+    ) -> iota_types::storage::error::Result<()> {
+        unimplemented!()
+    }
+
+    fn insert_committee(&self, _: Committee) -> iota_types::storage::error::Result<()> {
+        unimplemented!()
+    }
+}
+
+impl ReadStore for CheckpointSummaryFileStore<'_> {
```

Comment:
So if nothing is implemented then why are we even doing this?

Reply:
I feel you, and I hate it. However, this might not be the right PR to change a crate that we don't have ownership over. The API I am using requires me to implement those traits despite it being just a proxy to write those checkpoints to the filesystem. I would create a separate issue for this with a feature request to infra to provide such a method, or come up with something else tbh.

Reply:
It's `WriteStore : ReadStore` and  `ReadStore : ObjectStore`, and those none of those traits provide default implementations.

Reply:
But isn't it inherently wrong if it will fail any time the parent trait assumes the child implementation exists?